Docs: update sharing (#8239)
readthedocs/readthedocs.org: docs/commercial/sharing.rst

* by sending them a *secret link*,
* by giving them a *password*.

> These methods will allow them to view specific projects and versions inside your company.

Reviewer: ```suggestion
These methods will allow them to view specific projects or versions of a project inside your organization.
```

Additionally, you can use a HTTP Authorization Header.
This is useful to have access from a script.

Enabling
--------

> * Go into your *Project Admin* page and click on :guilabel:`Sharing`.

Reviewer: ```suggestion
* Go into your project's :guilabel:`Admin` page and click on :guilabel:`Sharing`.
```

To keep it consistent looking.

* Click on :guilabel:`New Share`
* Select access type (secret link, password, or HTTP header token),
  add an expiration date and a *Description* so you remember who you're sharing it with.

> * Check ``Allow access to all versions?`` if you want to grant access to all versions,

Reviewer 2: Shouldn't we use `:guilabel:` here?
